packages/headless/src/app/headless-engine.ts

```diff
@@ -106,6 +112,14 @@ export interface HeadlessConfigurationOptions {
    * A function that fetches a new access token. The function must return a Promise that resolves to a string (the new access token).
    */
   renewAccessToken?: () => Promise<string>;
+  /**
+   * Allows for augmenting a Platform request before it is sent.
+   * @param request Request to be augmented
+   * @param clientOrigin The origin of the client, can be "analyticsFetch", "analyticsBeacon" or "searchApiFetch"
```

Comment:
When calling the searchapi, should we distinguish what endpoint we are calling?
e.g. search vs. facetSearch vs. querySuggest?

Atm it seems like all of these are grouped under `searchApiFetch`?

Reply:
yes because it's the client origin/type, it's not scoping at the url or method.
Either checking the url, or us adding a preprocess for the search client payload which a user could use for different requests, like we do for different analytics events, would be the way to go.